### PR TITLE
supply points cases in SQL

### PR DESCRIPTION
**Pull request by snopoke**
@czue 
This ended up being a bit of a rabbit hole but somewhat necessary.

### CODE REVIEW

File: corehq/apps/users/models.py [hunk 1]
```diff
@@ -17,10 +17,11 @@
 from django.template.loader import render_to_string
 from django.utils.translation import ugettext as _
 from corehq.apps.app_manager.const import USERCASE_TYPE
-from corehq.apps.commtrack.dbaccessors import get_supply_point_case_by_location
 from corehq.apps.hqcase.dbaccessors import get_case_ids_in_domain_by_owner
 from corehq.apps.sofabed.models import CaseData
 from corehq.elastic import es_wrapper
+from corehq.form_processor.interfaces.processor import FormProcessorInterface
```
Comment (dimagimon) [inline corehq/apps/users/models.py]: F401 'FormProcessorInterface' imported but unused

File: corehq/form_processor/backends/couch/supply.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import logging
+
+from corehq.apps.commtrack.dbaccessors import get_supply_point_case_by_location
+from corehq.apps.commtrack.helpers import make_supply_point
+from corehq.form_processor.abstract_models import AbstractSupplyInterface
+from corehq.form_processor.models import CommCareCaseSQL
```
Comment (dimagimon) [inline corehq/form_processor/backends/couch/supply.py]: F401 'CommCareCaseSQL' imported but unused

File: corehq/form_processor/models.py [hunk 3]
```diff
@@ -347,6 +347,7 @@ class CommCareCaseSQL(PreSaveHashableMixin, models.Model, RedisLockableMixIn,
     deleted = models.BooleanField(default=False, null=False)
 
     external_id = models.CharField(max_length=255)
+    location_uuid = models.CharField(max_length=255, null=True)
```
Comment (snopoke) [inline corehq/form_processor/models.py]: This should really be a UUID field but can't be until we upgrade postgresql.

**Comment by czue** [comment]
looks good - not mergeable though

**Comment by czue** [comment]
build fails look real.
```
======================================================================
ERROR: testCreateSupplyPointFromFacility (custom.openlmis.tests.test_facility_sync.FacilitySyncTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/custom/openlmis/tests/test_facility_sync.py", line 43, in testCreateSupplyPointFromFacility
    self.assertEqual(sp1.location._id, loc1._id)
AttributeError: 'CommCareCase' object has no attribute 'location'
----------------------------------------------------------------------
```

**Comment by dimagimon** [comment]
There are 6 errors:

* corehq/apps/commtrack/tests/test_stock_state.py, line 178 - E501 line too long (120 > 115 characters)
* corehq/apps/commtrack/tests/test_stock_state.py, line 184 - E501 line too long (120 > 115 characters)
* corehq/apps/commtrack/tests/test_stock_state.py, line 190 - E501 line too long (120 > 115 characters)
* custom/ilsgateway/tanzania/warehouse/updater.py, line 467 - E501 line too long (120 > 115 characters)
* custom/ilsgateway/tanzania/warehouse/updater.py, line 470 - E501 line too long (118 > 115 characters)
* custom/ilsgateway/tanzania/warehouse/updater.py, line 586 - E501 line too long (120 > 115 characters)

**Comment by dimagimon** [comment]
There are 6 errors:

* corehq/apps/commtrack/tests/test_stock_state.py, line 178 - E501 line too long (120 > 115 characters)
* corehq/apps/commtrack/tests/test_stock_state.py, line 184 - E501 line too long (120 > 115 characters)
* corehq/apps/commtrack/tests/test_stock_state.py, line 190 - E501 line too long (120 > 115 characters)
* custom/ilsgateway/tanzania/warehouse/updater.py, line 467 - E501 line too long (120 > 115 characters)
* custom/ilsgateway/tanzania/warehouse/updater.py, line 470 - E501 line too long (118 > 115 characters)
* custom/ilsgateway/tanzania/warehouse/updater.py, line 586 - E501 line too long (120 > 115 characters)

**Comment by snopoke** [comment]
:white_check_mark:

**Comment by czue** [comment]
thanks for this! valiant effort!

**Comment by czue** [comment]
commtrack test coverage is definitely not as robust as forms (though is pretty decent). any reason to think this needs any additional scrutiny/QA? if not I'm good to merge

**Comment by snopoke** [comment]
probably good to get a bit of extra QA. What would you suggest?

**Comment by czue** [comment]
i'm honestly not sure. this might be one of those situations where no QA is going to catch anything that the tests wouldn't catch, so I may reverse my position. maybe can chat quickly tomorrow.

**Comment by dimagimon** [comment]
There are 6 errors:

* corehq/apps/commtrack/tests/test_stock_state.py, line 178 - E501 line too long (120 > 115 characters)
* corehq/apps/commtrack/tests/test_stock_state.py, line 184 - E501 line too long (120 > 115 characters)
* corehq/apps/commtrack/tests/test_stock_state.py, line 190 - E501 line too long (120 > 115 characters)
* custom/ilsgateway/tanzania/warehouse/updater.py, line 467 - E501 line too long (120 > 115 characters)
* custom/ilsgateway/tanzania/warehouse/updater.py, line 470 - E501 line too long (118 > 115 characters)
* custom/ilsgateway/tanzania/warehouse/updater.py, line 586 - E501 line too long (120 > 115 characters)

**Comment by snopoke** [comment]
I've done an 'audit' review of this and I'm pretty confident in the changes.

**Comment by czue** [comment]
:+1: yeah I think it's g2g. will merge when tests pass

**Comment by czue** [comment]
:boom:

**Comment by benrudolph** [comment]
:bear:

**Pull request merged**